libraries/transport/transportControl/transportControlI.H

```diff
@@ -108,13 +111,52 @@ void Foam::Pmt::transportControl<Base>::update()
 {
     if (!upToDate_)
     {
-        scalar maxDeltaC;
+        /*scalar maxDeltaC;
         bool maxDeltaCSet = this->controlDict().readCheckIfPresent
             (
                 "maxDeltaC", 
                 maxDeltaC,
                 [](scalar v){ return v>=0; }
             );
+        */
+
+        bool maxDeltaCSet = false;
+        List<scalar> maxDeltaC(composition_.species().size(), std::numeric_limits<scalar>::infinity());
+
+        const dictionary& maxDeltaCDict = this->controlDict().subOrEmptyDict("maxDeltaCValues");
+
+        if (maxDeltaCDict.size() >= 1){
+            maxDeltaCSet = true;
+            for (const auto& maxDCSpecies : maxDeltaCDict)
+            {
+                const auto speciesName = maxDCSpecies.dict().dictName();
+                const label speciesIndex = composition_.species().find(speciesName);
+
+                if (speciesIndex<0)
+                {
+                    FatalErrorInFunction
+                        << "Sub-dict maxDeltaCValues in controlDict has an unknown species." << nl
+                        << abort(FatalError);
+                }
+                
+                maxDeltaC[speciesIndex] = dimensionedScalar("maxDeltaC", dimless, maxDCSpecies.dict()).value();
```

Comment:
I don't think `controlDict` contain `dimensionedScalar`s by convention (and if it did it would have to have the dimensions of the concentration fields). Could we change this into a simple `scalar` read?